authorisation.cpp

```diff
@@ -182,13 +182,28 @@ bool DeRestPluginPrivate::checkApikeyAuthentification(const ApiRequest &req, Api
                 }
             }
 
+            if ((!(i->useragent.isEmpty()) && i->useragent.startsWith(QLatin1String("iConnect"))) || i->devicetype.startsWith(QLatin1String("iConnectHue")))
+            {
+                req.mode = ApiModeStrict;
+            }
+            else if (i->devicetype.startsWith(QLatin1String("Echo")))
+            {
+                req.mode = ApiModeEcho;
+            }
+            else if (i->devicetype.startsWith(QLatin1String("hue_")) ||
+                     i->devicetype.startsWith(QLatin1String("Hue ")))
```

Comment:
See comment: https://github.com/dresden-elektronik/deconz-rest-plugin/pull/1046#issuecomment-451887310

Reply:
@Thomas-Vos Should work again with 2.05.55.

Another thing I've noticed, the Phoscon App creates resourcelinks which contain `content like this:
```
{"classid":2020,"description":null,"links":["/userparameter/2","/groups/3","/sensors/29","/sensors/49","/sensors/50","/rules/22","/rules/15","/rules/18","/rules/5","/rules/24","/rules/23","/rules/19","/rules/16","/rules/8","/rules/42"],"name":"Fluri3","owner":"4E8BDC7B8E","recycle":false,"type":"Link"}
```
Here are two problems:
1.  `description` should never be `null` but an empty string: this is fixed in 2.05.55
2. The /userparameter/2  in links[] isn't allowed by Hue Essentials and an error is thrown. This can only be addressed in the Hue Essentials app.

Reply:
@manup Thanks for your change. I have addressed your points in the upcoming release 1.1.0 which will be available soon.